Makefile

```diff
@@ -2,8 +2,8 @@ CC := g++
 MAJOR_VERSION := 1
 MINOR_VERSION := 0
 MICRO_VERSION := 0
-CFLAGS := -std=gnu++14 -Wall -Wextra -O3 -g -fno-strict-aliasing
-LDFLAGS :=
+CFLAGS := -std=gnu++14 -Wall -Wextra -O3 -g -fno-strict-aliasing `python-config --cflags`
```

Comment:
I didn't know about `python-config`, this looks like it simplifies a lot of the config we have with `which_py_include` and `which_py_so`. Since this is already passing that information, should we drop our custom scripts? Also, I get a warning from gcc when using these flags:

```
cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
```

I think we could that flag out or something.